utils/CI/schema_validation.sh

```diff
@@ -80,6 +80,9 @@ validate_campaign() {
 
 RET=0
 
+# remove any_tag to actually see errors in action WML
+patch -p 1 < utils/CI/schema_validation.patch || RET=1
```

Comment:
Likely to get a "fails to build from source twice in a row" error from Debian, if they get round to enabling all tests.

I feel we should either merge it, or leave it as a PR that people can merge with Git, but not carry it as a patch.

Reply:
If you expect random people to run CI scripts and get confused by the changes we can certainly revert the patch at the end. IMO it’s easier the way I set it up here but I don’t mind much either way.

If you want to propose removing the any tag stuff feel free. As long as no one does that it’s useful to at least remove it for the CI check so we actually notice issues.

Reply:
I'm not sure if `any_tag` is actually needed; it might be equivalent to just defining a tag with `name="*"`. But _conceptually_ it's definitely needed – in general, the WML schema allows anything to go in the places where it's used. We shouldn't make the schema reject valid WML in someone's add-on.

There is definitely room for adding "extension hooks" so that add-ons – and the mainline campaigns for that matter – can define tags that will be "included" into specific points of the main schema. In the long run, that is probably better than this; but this seems reasonable in the short term.

Reply:
I'd prefer using the preprocessor over patching.

Reply:
You mean like adding

```
#ifdef INCLUDE_MAINLINE_EXTENSIONS
# tag definitions
#else
any_tag=true
#endif
```

directly in the schema file?

I'm not too sure about that… to me it seems messier than the patch version…

Reply:
Currently there is no way to influence how schema files are preprocessed. If you're thinking about `--preprocess-defines` then that defines symbols for preprocessing the files that get validated.

Reply:
Not quite on-topic here but it might be reasonable to change it so that `--preprocess-defines` influences
https://github.com/wesnoth/wesnoth/blob/14df301b7f9a2f5457f1346577e7ff7ce7590e43/src/config_cache.cpp#L42-L53
Then those symbols would perhaps be defined "everywhere".